File: tmdb-core/src/main/java/dev/reuss/tmdb/value/id/ReviewId.java
package dev.reuss.tmdb.value.id;

/**
 * Strongly typed identifier for a TMDB review resource.
 *
 * <p>TMDB review ids are string values.</p>
 *
 * @param value the raw TMDB review id
 */
public record ReviewId(String value) implements TmdbResourceId {

    public ReviewId {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("Review id must not be blank");
        }

        value = value.trim();
    }

    public static ReviewId of(String value) {
        return new ReviewId(value);
    }

    @Override
    public String asString() {
        return value;
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/value/id/ReviewId.kt
package dev.reuss.tmdb.value.id

/**
 * Strongly typed identifier for a TMDB review resource.
 *
 * TMDB review ids are string values.
 *
 * @property value the raw TMDB review id
 */
class ReviewId private constructor(
    value: String,
) : StringTmdbResourceId(value, "Review id") {
    companion object {
        @JvmStatic
        fun of(value: String): ReviewId = ReviewId(value)
    }
}
